File: java/oop/AccountAppWithException/src/com/techlabs/account/InsufficientFundException.java
package com.techlabs.account;

public class InsufficientFundException extends Exception {

	/**
	 * 
	 */
	private static final long serialVersionUID = -7784493461754864709L;
	private String string;
	private Account account=null;
	public InsufficientFundException(String string){
		this.string=string;
	}
	
	public InsufficientFundException(Account account,String string){
		this(string);
		this.account=account;
	}
	
	@Override
	public String getMessage(){
		if(this.account!=null)
			return this.string+"\n"+account;
		return this.string;
	}
}

File: dotNet/CSharp/AccountAppWithCustomException/AccountAppWithCustomException/InsufficientFundException.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AccountAppWithCustomException
{
    class InsufficientFundException :Exception
    {
        private string _message;
        private Account _account = null;

        public InsufficientFundException(string message,Account account) : this(message)
        {
            this._account = account;
        }

        public InsufficientFundException(string message)
        {
            this._message = message;
        }

        public override string Message
        {
            get
            {
                if (this._account!=null)
                {
                    return this._message + "\n" + this._account;
                }
                return this._message;
            }
        }
    }
}
